backend/src/impl/default_controllers_impl.py

```diff
@@ -128,3 +150,120 @@ def systems_system_id_delete(system_id: str):
+
+
+def systems_analyses_get(system_ids_str: str, pairwise_performance_gap: str):
+    if not is_boolean_string(pairwise_performance_gap):
+        abort_with_error_message(
+            400,
+            "pairwise_performance_gap is not a boolean string: "
+            + f"{pairwise_performance_gap}",
+        )
+    do_pairwise_performance_gap = string_to_boolean(pairwise_performance_gap)
+
+    single_analyses: dict = {}
+    system_ids: list = system_ids_str.split(",")
+    system_name = None
+    task = None
+    creator = None
+    page = 0
+    page_size = len(system_ids)
+    sort = None
+    systems: list[System] = SystemModel.find(
+        system_ids,
+        page,
+        page_size,
+        system_name,
+        task,
+        sort,
+        creator,
+        include_metric_stats=True,
+    ).systems
+    systems_len = len(systems)
+    if systems_len == 0:
+        return SystemAnalysesReturn(single_analyses)
+
+    if do_pairwise_performance_gap and systems_len != 2:
+        abort_with_error_message(
+            400,
+            "pairwise_performance_gap=true"
+            + f" only accepts 2 systems, got: {systems_len}",
+        )
+
+    for system in systems:
+        system_info: SystemInfo = system.system_info
+        system_info_dict = system_info.to_dict()
+        system_output_info = SysOutputInfo.from_dict(system_info_dict)
+
+        # TODO(chihhao) bug in SDK
+        # nested dataclasses must be from_dict()ed properly
+        # move this function to SDK in the future or
+        # add proper from_dict() methods in SDK
+        for feature_name, feature in system_output_info.features.items():
+            bucket_info = feature["bucket_info"]
+            if bucket_info is not None:
+                bucket_info = BucketInfo(**bucket_info)
+
+            # TODO Feature is not getting from_dict()ed properly, hardcode for now:
+            _type = feature["_type"]
+            feature.pop("_type")
+            if _type == ClassLabel._type:
+                feature = ClassLabel(**feature)
+            elif _type == Sequence._type:
+                feature = Sequence(**feature)
+            elif _type == Set._type:
+                feature = Set(**feature)
+            elif _type == Position._type:
+                feature = Position(**feature)
+            elif _type == Span._type:
+                feature = Span(**feature)
+            elif _type == Value._type:
+                feature = Value(**feature)
+
+            feature.bucket_info = bucket_info
+            system_output_info.features[feature_name] = feature
+
+        system_output_info.results = Result(**system_output_info.results)
+        # The order of getting the processor first then setting
+        # the tokenizer is unnatural, but in the SDK get_default_tokenizer
+        # relies on the processor's TaskType inforamtion
+
+        processor = get_processor(system_output_info.task_name)
+        system_output_info.tokenizer = get_default_tokenizer(
```

Comment:
`system_output_info.task_name` should be the str version of `processor.task_type`(enum). I think you can do `TaskType(task_name)` to turn the str version to the enum.